### ISSUE TITLE
Allow mismatch in server and client side rendered HTML per component

### ISSUE THREAD

**Issue by pawelgur** [nuxt/nuxt#9441]
### Is your feature request related to a problem? Please describe.

I have two versions of responsive main menu component, e.g. `<main-menu-desktop>` and `<main-menu-mobile>`, decision which to render is based on screen width, e.g. breakpoint X. 

This is SEO critical component that must be server side rendered. 
Both components are heavy, thus can't render both (one hidden with css) due to performance constraints.
Both components can't be made same due to business requirements.
There is no 100% reliable client's screen width detection mechanism available on server (maybe until [client hints](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/client-hints) are widely supported), thus there is always a possibility of mismatch between client and server.

In such case we are making conscious decision to always render mobile version on server and replace it with desktop component on client if needed (based on actual viewport width).

**Problem**: in dev mode Vue throws "The client-side rendered virtual DOM tree is not matching server-rendered content." warning and bails hydration. There is no (known to me) way of disabling it.

Even though this is framework's sensible default behavior, but there should be opt-out option to be able to cover edge case mentioned above.

### Describe the solution you'd like

Create a feature that would disable warning and hydration bailing in dev mode for selected components.

E.g. 
```
<dynamic-ssr>
    <main-menu-desktop v-if="showDesktopBasedOnBreakpointOrFallback" />
    <main-menu-mobile v-else />
</dynamic-ssr>
```

Expectation: component is server side rendered and no DOM mismatch warning and full rehydration during dev build

Basically same behavior like on prod build (just forced explicitly)

### Describe alternatives you've considered

There are plenty of threads related to this warning, but solution in most of them is to not server-side render such components.
This solution is not applicable in my case due to SEO requirement. 

Other suggested solutions are rendering both components and hiding with css (v-show), which is also not an option due to performance constraints

https://github.com/nuxt/nuxt.js/issues/8266 is similar issue, but author went into direction of forced full rehydration, which is not   desired due to performance hit.  What I'd like to have is kind of forced rehydration localized to small part of DOM.

### Additional context

Any suggestions welcome, including temporary hacks.

**Comment by pawelgur** [comment]
Another more simple use case: **placeholder component for data fetched on client site**  
```
<some-component v-if="data" />
<placeholder-component v-else />
```
During server side rendering data is not available and placeholder is rendered. When data is fetched on client-side before hydration (e.g. preloaded or hydration is delayed with vue-lazy-hydration), this will result in state mismatch and hydration error.

Hydration mismatch in such case is expected and developer should have a way to suppress hydration error.

**Comment by danielroe** [comment]
We are approaching the Nuxt 2 EOL date (June 30, 2024) - see [this article](https://nuxt.com/blog/nuxt2-eol) for more information. I'm closing this issue as it's marked as a _Nuxt 2 related enhancement_ and it's not a critical issue.

That doesn't mean it might not be relevant for Nuxt 3. If it is, please feel free to open a new issue (or just comment, and I can reopen it). 🙏

Thank you for your understanding and for your contribution to Nuxt! 🎉
